[V3 Downloader] Error from invalid characters in repo name

# Command bugs

#### Command name

`[p]repo add`

#### What cog is this command from?

Downloader

#### What were you expecting to happen?

A warning message that the repo name is invalid when naming a repo something with invalid characters (such as `:` in windows)

#### What actually happened?

Error:
```py
[2019-09-14 10:39:30] [ERROR] red: Exception in command 'repo add'
Traceback (most recent call last):
  File "<VENV>\lib\site-packages\redbot\cogs\downloader\downloader.py", line 233, in _repo_add
    repo = await self._repo_manager.add_repo(name=name, url=repo_url, branch=branch)
  File "<VENV>\lib\site-packages\redbot\cogs\downloader\repo_manager.py", line 586, in add_repo
    r = Repo(url=url, name=name, branch=branch, folder_path=self.repos_folder / name)
  File "<VENV>\lib\site-packages\redbot\cogs\downloader\repo_manager.py", line 66, in __init__
    self.folder_path.mkdir(parents=True, exist_ok=True)
  File "<Python>\lib\pathlib.py", line 1241, in mkdir
    self._accessor.mkdir(self, mode)
NotADirectoryError: [WinError 267] The directory name is invalid: '<datapath>\\cogs\\RepoManager\\repos\\:'
```

#### How can we reproduce this issue?

Run `[p]repo add : aaaaa`
Type "I agree" if prompted

--- a/changelog.d/downloader/2827.bugfix.rst
+++ b/changelog.d/downloader/2827.bugfix.rst
@@ -1,1 +1,1 @@
-Invalid repo names no longer cause an error.
+Repo names can now only contain the characters listed in the help text (A-Z, 0-9, underscores, and hyphens).

--- a/changelog.d/downloader/2827.bugfix.rst
+++ b/changelog.d/downloader/2827.bugfix.rst
@@ -1,1 +1,1 @@
-Repo names can now only contain the characters listed in the help text (A-Z, 0-9, and underscores).
+Invalid repo names no longer cause an error.

--- a/changelog.d/downloader/2827.bugfix.rst
+++ b/changelog.d/downloader/2827.bugfix.rst
@@ -1,0 +1,1 @@
+Repo names can now only contain the characters listed in the help text (A-Z, 0-9, and underscores).

--- a/redbot/cogs/downloader/downloader.py
+++ b/redbot/cogs/downloader/downloader.py
@@ -1,6 +1,7 @@
 import asyncio
 import contextlib
 import os
+import re
 import shutil
 import sys
 from pathlib import Path
@@ -222,11 +223,16 @@ class Downloader(commands.Cog):
     async def _repo_add(self, ctx, name: str, repo_url: str, branch: str = None):
         """Add a new repo.
 
-        We recommend using only A-z, numbers and underscores in the repo name for compatibility.
+        Repo names can only contain characters A-z, numbers, underscores, and hyphens.
         The branch will be the default branch if not specified.
         """
         agreed = await do_install_agreement(ctx)
         if not agreed:
+            return
+        if re.match("^[a-zA-Z0-9_\-]*$", name) is None:
+            await ctx.send(
+                _("Repo names can only contain characters A-z, numbers, underscores, and hyphens.")
+            )
             return
         try:
             # noinspection PyTypeChecker

--- a/redbot/cogs/downloader/downloader.py
+++ b/redbot/cogs/downloader/downloader.py
@@ -1,6 +1,7 @@
 import asyncio
 import contextlib
 import os
+import re
 import shutil
 import sys
 from pathlib import Path
@@ -227,6 +228,11 @@ class Downloader(commands.Cog):
         """
         agreed = await do_install_agreement(ctx)
         if not agreed:
+            return
+        if re.match("^[a-zA-Z0-9_]*$", name) is None:
+            await ctx.send(
+                _("Repo names can only contain characters A-z, numbers and underscores.")
+            )
             return
         try:
             # noinspection PyTypeChecker

--- a/redbot/cogs/downloader/downloader.py
+++ b/redbot/cogs/downloader/downloader.py
@@ -243,7 +243,9 @@ class Downloader(commands.Cog):
             )
         except OSError:
             await ctx.send(
-                _("Something went wrong trying to add that repo. Your repo name might have an invalid character.")
+                _(
+                    "Something went wrong trying to add that repo. Your repo name might have an invalid character."
+                )
             )
         else:
             await ctx.send(_("Repo `{name}` successfully added.").format(name=name))

--- a/redbot/cogs/downloader/downloader.py
+++ b/redbot/cogs/downloader/downloader.py
@@ -1,7 +1,6 @@
 import asyncio
 import contextlib
 import os
-import re
 import shutil
 import sys
 from pathlib import Path
@@ -223,16 +222,11 @@ class Downloader(commands.Cog):
     async def _repo_add(self, ctx, name: str, repo_url: str, branch: str = None):
         """Add a new repo.
 
-        The name can only contain characters A-z, numbers and underscores.
+        We recommend using only A-z, numbers and underscores in the repo name for compatibility.
         The branch will be the default branch if not specified.
         """
         agreed = await do_install_agreement(ctx)
         if not agreed:
-            return
-        if re.match("^[a-zA-Z0-9_]*$", name) is None:
-            await ctx.send(
-                _("Repo names can only contain characters A-z, numbers and underscores.")
-            )
             return
         try:
             # noinspection PyTypeChecker
@@ -246,6 +240,10 @@ class Downloader(commands.Cog):
                 repo_url,
                 branch,
                 exc_info=err,
+            )
+        except OSError:
+            await ctx.send(
+                _("Something went wrong trying to add that repo. Your repo name might have an invalid character.")
             )
         else:
             await ctx.send(_("Repo `{name}` successfully added.").format(name=name))

